napari/utils/_pydantic/evented_model.py

```diff
@@ -0,0 +1,67 @@
+from typing import Dict
+
+from pydantic import BaseModel, PrivateAttr
+
+from ..events.dataclass import _type_to_compare, is_equal
+from ..events.event import EmitterGroup
+from .custom_types import JSON_ENCODERS
+
+
+class EventedModel(BaseModel):
+
+    # add private attributes for event emission
+    _events: EmitterGroup = PrivateAttr(default_factory=EmitterGroup)
+    __equality_checks__: Dict = PrivateAttr(default_factory=dict)
+
```

Comment:
```suggestion
    __slots__: ClassVar[Set[str]] = {"__weakref__"}

```

Reply:
(see below... it only seemed to be ok without this because your EmitterGroup wasn't getting assigned a source)

Reply:
Do we need a `source`? I guess might as well ... ??

Reply:
Yeah, i think we definitely dont want to lose that.  Things like nested evented list take good advantage of that attribute...

Reply:
sounds good, thanks for the fix!